arbnode/simple_redis_lock.go

```diff
@@ -0,0 +1,210 @@
+package arbnode
+
+import (
+	"context"
+	"crypto/rand"
+	"math"
+	"math/big"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/go-redis/redis/v8"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/pkg/errors"
+	flag "github.com/spf13/pflag"
+)
+
+type SimpleRedisLock struct {
+	stopwaiter.StopWaiter
+	client      redis.UniversalClient
+	config      *SimpleRedisLockConfig
+	lockedUntil int64
+	mutex       sync.Mutex
+	stopping    bool
+	readyToLock func() bool
+	myId        string
+}
+
+type SimpleRedisLockConfig struct {
+	RedisUrl        string        `koanf:"redis-url"`
+	MyId            string        `koanf:"my-id"`
+	LockoutDuration time.Duration `koanf:"lockout-duration"`
+	RefreshDuration time.Duration `koanf:"refresh-duration"`
+	Key             string        `koanf:"key"`
+	BackgroundLock  bool          `koanf:"background-lock"`
+}
+
+func RedisLockConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.String(prefix+".redis-url", DefaultRedisLockConfig.RedisUrl, "URL of redis server for coordination")
+	f.String(prefix+".my-id", DefaultRedisLockConfig.RedisUrl, "this node's id prefix when acquiring the lock (optional)")
+	f.Duration(prefix+".lockout-duration", DefaultRedisLockConfig.LockoutDuration, "how long lock is held")
+	f.Duration(prefix+".refresh-duration", DefaultRedisLockConfig.RefreshDuration, "how long between consecutive calls to redis")
+	f.String(prefix+".key", prefix+".simple-lock-key", "key for lock")
```

Comment:
Ideally as used in the batch poster the key and redis url be the same as the data poster. I'm not exactly sure how to organize the config for that, but maybe we could have the data poster grab these config values from the redis lock?

Reply:
Redis URL should probably be shared, and can get a separate config struct under node.
Encryption key - I'm not sure if should be shared.. that's worth some thought.

Reply:
Re: sharing the key, my thinking is that the batch poster lock and the data poster redis storage will be enabled on exactly the same nodes, so it makes sense to use the same key for both (since any node which had one key would need to have the other).

Reply:
Nevermind on this, I was confusing the signing key vs the redis db key. We'll just want to share the URL.